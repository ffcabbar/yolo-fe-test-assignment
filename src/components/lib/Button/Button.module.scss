.button {
  background: #ff4700;
  color: white;
  border-radius: 2rem;
  border: none;
  margin: 0;
  width: 100%;
  padding: 0.8rem 1rem;
  cursor: pointer;
  transition-duration: 0.2s;
  transition-timing-function: cubic-bezier(0.2, 0.8, 0.4, 1);

  &:hover {
    background: darken(#ff4700, 5%);
    transform: translateY(-2px);
  }
}
