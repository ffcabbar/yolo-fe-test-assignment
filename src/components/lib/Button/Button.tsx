import styles from './Button.module.scss';

type ButtonProps = {
  children?: React.ReactNode;
};

type A = React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>;

type Props = ButtonProps & A;

export const Button = ({ children, ...rest }: Props) => {
  return (
    <button className={styles.button} {...rest}>
      {children}
    </button>
  );
};
