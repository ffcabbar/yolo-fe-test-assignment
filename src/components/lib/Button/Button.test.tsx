import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './Button';
import renderer from 'react-test-renderer';

test('renders button', () => {
  render(<Button children="Yolo Group" />);
  const buttonElement = screen.getByText(/Yolo Group/i);
  expect(buttonElement).toBeInTheDocument();
});

test('handles onClick', () => {
  const onClick = jest.fn();
  render(<Button onClick={onClick} children="Yolo Group" />);
  const buttonElement = screen.getByText('Yolo Group');
  fireEvent.click(buttonElement);
  expect(onClick).toHaveBeenCalledTimes(1);
});

test('matches snapshot', () => {
  const tree = renderer.create(<Button children="Yolo Group" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('matches snapshot correctly when there are no props', () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});
