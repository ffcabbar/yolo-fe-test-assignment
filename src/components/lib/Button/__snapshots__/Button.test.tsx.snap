// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="button"
>
  Yolo Group
</button>
`;

exports[`matches snapshot correctly when there are no props 1`] = `
<button
  className="button"
/>
`;
